source/Calamari/Kubernetes/Authentication/AwsCliAuth.cs

```diff
@@ -0,0 +1,134 @@
+                var awsCliVersion = awsCli.GetAwsCliVersion();
+                var minimumAwsCliVersionForAuth = new SemanticVersion("1.16.156");
+                if (awsCliVersion.CompareTo(minimumAwsCliVersionForAuth) > 0)
+                {
+                    var region = GetEksClusterRegion(clusterUrl);
+                    if (!string.IsNullOrWhiteSpace(region))
+                    {
+                        SetKubeConfigAuthenticationToAwsCli(user, clusterName, region);
+                        return true;
+                    }
+
+                    log.Verbose("The EKS cluster Url specified should contain a valid aws region name");
+                }
+
+                log.Verbose(
+                    $"aws cli version: {awsCliVersion} does not support the \"aws eks get-token\" command. Please update to a version later than 1.16.156");
+            }
+            catch (Exception e)
+            {
+                log.Verbose(
+                    $"Unable to authenticate to {clusterUrl} using the aws cli. Failed with error message: {e.Message}");
+            }
+
+            return false;
+        }
+
+        void SetKubeConfigAuthenticationToAwsIAm(string user, string clusterName)
+        {
+            var kubectlVersion = kubectl.GetVersion();
+            var apiVersion = kubectlVersion.Some() && kubectlVersion.Value > new SemanticVersion("1.23.6")
+                ? "client.authentication.k8s.io/v1beta1"
+                : "client.authentication.k8s.io/v1alpha1";
+
+            kubectl.ExecuteCommandAndAssertSuccess(
+                "config",
+                "set-credentials",
+                user,
+                "--exec-command=aws-iam-authenticator",
+                $"--exec-api-version={apiVersion}",
+                "--exec-arg=token",
+                "--exec-arg=-i",
+                $"--exec-arg={clusterName}");
+        }
+
+        void ConfigureAwsCli()
+        {
+            if (!environmentVars.ContainsKey("AWS_ACCESS_KEY_ID"))
+            {
+                var awsEnvironmentGeneration =
+                    AwsEnvironmentGeneration.Create(log, deploymentVariables).GetAwaiter().GetResult();
+                environmentVars.AddRange(awsEnvironmentGeneration.EnvironmentVars);
+            }
+        }
+
+        string GetEksClusterRegion(string clusterUrl) => clusterUrl.Replace(".eks.amazonaws.com", "").Split('.').Last();
+
+        void SetKubeConfigAuthenticationToAwsCli(string user, string clusterName, string region)
+        {
+            var token = deploymentVariables.Get(AccountVariables.Jwt);
+
+            if (string.IsNullOrEmpty(token))
+            {
+                token = awsCli.GetEksClusterToken(clusterName, region);
+            }
+
+            var arguments = new List<string> { "config", "set-credentials", user, $"--token={token}" };
+
+            log.AddValueToRedact(token, "<token>");
+            kubectl.ExecuteCommandAndAssertSuccess(arguments.ToArray());
+        }
```

Comment:
This is where we change to now set the token statically instead of configuring it to grab the token as part of running `kubectl`